reporter/otlp_reporter.go

```diff
@@ -574,7 +578,9 @@ func (r *OTLPReporter) getProfile() (profile *profiles.Profile, startTS, endTS u
 					fileIDtoMapping[traceInfo.files[i]] = idx
 					locationMappingIndex = idx
 
-					execInfo, exists := r.executables.Get(traceInfo.files[i])
+					// Ensure that actively used executables do not expire.
+					execInfo, exists := r.executables.GetAndRefresh(traceInfo.files[i],
+						executableCacheLifetime)
```

Comment:
What is the difference here? Does not `Get` automatically reset the TTL? If not, the other usage of the `lru` should be reviewed and this same change probably applies to (potential multiple) other locations.

Reply:
`Get` updates the recentness of an item, which covers eviction when the LRU is full.
The lifetime is only updated/set with `Add` and is not touched by `Get`.
The lifetime is a concept that allows purging unused items from an LRU (call to `PurgeExpired`) to avoid long-term resource leaks.

`GetAndRefresh`, updates both, recentness and lifetime of an item.

FreeLRU does not actively purge expired items. For the reporter caches, we recently added a ticker to call to `PurgeExpired` regularly. For other caches we don't have an active purging.